File: all-submissions/0118-pascals-triangle/solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        
        for(int i = 0; i<numRows; i++)
        {
            List<Integer> temp = new ArrayList<>();
            for(int j = 0; j<=i; j++)
            {
                if(i==j||j==0)
                    temp.add(1);
                else
                {
                   int a = triangle.get(i-1).get(j-1) + triangle.get(i-1).get(j);
                   temp.add(a);
                }
            }
            triangle.add(temp);
        }
        return triangle;
    }
}

File: all-submissions/0118-pascals-triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = [[1]]
        if numRows == 1:
            return ans
        for i in range(2, numRows + 1):
            res = []
            curr = ans[-1]
            res.append(1)
            for j in range(len(curr)-1):
                res.append(curr[j] + curr[j+1])
            res.append(1)
            ans.append(res)
        return ans


